planner/core/exhaust_physical_plans.go

```diff
@@ -440,36 +442,33 @@ func (p *LogicalJoin) constructIndexJoin(
 	}
 
 	var outerHashKeys, innerHashKeys []*expression.Column
-	// HashKey is only used for IndexJoin and IndexHashJoin since they need to
-	// build hash tables.
-	if joinTP != plancodec.TypeIndexMergeJoin {
```

Comment:
`joinTP` in the function definition can be removed.

Reply:
Done.